File: backend/src/main/java/com/festago/school/dto/SchoolsResponse.java
package com.festago.school.dto;

import static java.util.stream.Collectors.collectingAndThen;
import static java.util.stream.Collectors.toList;

import com.festago.school.domain.School;
import java.util.List;

/**
 * @deprecated API 버저닝이 적용되면 해당 클래스 삭제
 */
@Deprecated(forRemoval = true)
public record SchoolsResponse(
    List<SchoolResponse> schools) {

    public static SchoolsResponse from(List<School> schools) {
        return schools.stream()
            .map(SchoolResponse::from)
            .collect(collectingAndThen(toList(), SchoolsResponse::new));
    }
}

File: android/festago/app/src/main/java/com/festago/festago/data/dto/SchoolsResponse.kt
package com.festago.festago.data.dto

import com.festago.festago.model.School
import kotlinx.serialization.Serializable

@Serializable
data class SchoolsResponse(
    val schools: List<SchoolResponse>
) {
    fun toDomain(): List<School> = schools.map { it.toDomain() }
}
